File: src/java/src/main/java/com/btgpactualsolutions/etd/config/InitiatorSettings.java
package com.btgpactualsolutions.etd.config;

public class InitiatorSettings {
	private String senderCompID;
	private String targetCompID;
	private String socketConnectHost;
	private long socketConnectPort;
	private String sslCertificate;	
	private FixSettings fixSettings;

	public InitiatorSettings() {
		this.fixSettings = new FixSettings();
	}
	
    public String getSenderCompID() {
        return senderCompID;
    }

    public void setSenderCompID(String senderCompID) {
        this.senderCompID = senderCompID;
    }

    public String getTargetCompID() {
        return targetCompID;
    }

    public void setTargetCompID(String targetCompID) {
        this.targetCompID = targetCompID;
    }

    public String getSocketConnectHost() {
        return socketConnectHost;
    }

    public void setSocketConnectHost(String socketConnectHost) {
        this.socketConnectHost = socketConnectHost;
    }

    public long getSocketConnectPort() {
        return socketConnectPort;
    }

    public void setSocketConnectPort(long socketConnectPort) {
        this.socketConnectPort = socketConnectPort;
    }

    public String getSSLCertificate() {
        return sslCertificate;
    }

    public void setSSLCertificate(String sslCertificate) {
        this.sslCertificate = sslCertificate;
    }

    public FixSettings getFixSettings() {
    	return fixSettings;
    }
    
    public void setAppSettings(FixSettings app) {
        app = app != null ? app : new FixSettings();

        fixSettings.setReconnectInterval(app.getReconnectInterval());        
        fixSettings.setFileStorePath(app.getFileStorePath());
        fixSettings.setStartTime(app.getStartTime());
        fixSettings.setEndTime(app.getEndTime());
        fixSettings.setHeartBtInt(app.getHeartBtInt());
        fixSettings.setLogoutTimeout(app.getLogoutTimeout());
        fixSettings.setResetOnLogon(app.getResetOnLogon());
        fixSettings.setResetOnLogout(app.getResetOnLogout());
        fixSettings.setResetOnDisconnect(app.getResetOnDisconnect());        
        fixSettings.setBeginString(app.getBeginString());
        fixSettings.setDialect(app.getDialect());
        fixSettings.setSSLEnable(app.getSSLEnable());
        fixSettings.setSSLValidateCertificates(app.getSSLValidateCertificates());
    }
}

File: src/dotnet/src/btg-solutions-etd/core/settings/InitiatorSettings.cs
namespace BtgPactualSolutions.Etd.core.settings {
    public class InitiatorSettings {
        public InitiatorSettings() {
            FixSettings = new FixSettings();
        }

        public string SenderCompID { get; set; }
        public string TargetCompID { get; set; }
        public string SocketConnectHost { get; set; }
        public long SocketConnectPort { get; set; }
        public string SSLCertificate { get; set; }
        public FixSettings FixSettings { get; private set; }


        public void SetAppSettings(FixSettings app = null) {
            app = app ?? new FixSettings();

            FixSettings.ReconnectInterval = app.ReconnectInterval;
            FixSettings.FileStorePath = app.FileStorePath;
            FixSettings.FileLogPath = app.FileLogPath;
            FixSettings.StartTime = app.StartTime;
            FixSettings.EndTime = app.EndTime;
            FixSettings.HeartBtInt = app.HeartBtInt;
            FixSettings.LogoutTimeout = app.LogoutTimeout;
            FixSettings.ResetOnLogon = app.ResetOnLogon;
            FixSettings.ResetOnLogout = app.ResetOnLogout;
            FixSettings.ResetOnDisconnect = app.ResetOnDisconnect;
            FixSettings.Dialect = app.Dialect;
        }
    }
}
